src/lib/traced_data_csv_io.py

```diff
@@ -0,0 +1,43 @@
+import csv
+import time
+
+from src.lib import PipelineConfiguration
+from core_data_modules.cleaners import Codes
+from core_data_modules.traced_data import Metadata, TracedData
+
+_td_type_error_string = "argument 'data' contains an element which is not of type TracedData"
+
+class TracedDataCSVIO(object):
+    @staticmethod
+    def export_traced_data_iterable_to_csv(data, f, headers=None):
+        """
+        Writes a collection of TracedData objects to a CSV.
+        Columns will be exported in the order declared in headers if that parameter is specified,
+        otherwise the output order will be arbitrary.
+        :param data: TracedData objects to export.
+        :type data: iterable of TracedData
+        :param f: File to export to.
+        :type f: file-like
+        :param headers: Headers to export. If this is None, all headers will be exported.
+        :type headers: list of str
+        """
+        data = list(data)
+        for td in data:
+            assert isinstance(td, TracedData), _td_type_error_string
+
+        # If headers unspecified, search data for all headers which were used
+        if headers is None:
```

Comment:
(This shouldn't be possible given how it's used in pipelines, so you could remove make headers non-optional and delete this block)